### PR TITLE
Bug 1284403 - show error message dialog when clicking sync/sign-in under offline

### PR DESCRIPTION
**Pull request by danhuang1202**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d7cdd6560eb14634e881e5bc805c092c388be545

File: tv_apps/browser/js/browser_dialog.js [hunk 1]
```diff
@@ -222,6 +244,16 @@ var BrowserDialog = {
         });
         break;
 
+      case 'sync_error':
+        this.openAlertDialog({
+          titleL10nId: options.titleL10nId,
+          messageL10nId: options.messageL10nId,
```
Comment (weilonge) [inline tv_apps/browser/js/browser_dialog.js]: Please check if all L10N ID can be mapped into the translation file.

File: tv_apps/smart-system/js/sync_manager.js [hunk 2]
```diff
@@ -765,10 +765,10 @@
       port.postMessage(message);
     },
 
-    notifyStateChange() {
+    notifyStateChange(state) {
       this.managementPortMessage({
         name: 'onsyncchange',
-        state: this.state,
```
Comment (weilonge) [inline tv_apps/smart-system/js/sync_manager.js]: Is there any other case to use this.state and will cause other similar issues?

File: tv_apps/browser/js/browser_dialog.js (hunk 1 above)
Comment (danhuang1202) [inline tv_apps/browser/js/browser_dialog.js]: The fx-sync error l10n id only support 3 type of errors: ERROR_UNKNOWN, ERROR_INVALID_SYNC_ACCOUNT and ERROR_OFFLINE.
And these three l10n id are all support in translation file.

Please see the tv_apps/browser/js/sync/settings.js, line 234-244
```
var errorIds = [ERROR_UNKNOWN, ERROR_UNKNOWN + '-explanation'];

const KNOWN_ERRORS = [
    ERROR_INVALID_SYNC_ACCOUNT,
    ERROR_OFFLINE
];
```

File: tv_apps/smart-system/js/sync_manager.js (hunk 2 above)
Comment (danhuang1202) [inline tv_apps/smart-system/js/sync_manager.js]: There is no similar issue in sync_manager.js like this.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1ac9c5f726361ea6465c3a56c0f0c9a825694b85

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=75b7e22d2a04ba9d10284031f4d46a834e532b56

File: tv_apps/browser/js/sync/settings.js [hunk 3]
```diff
@@ -243,14 +243,14 @@
...
+            BrowserDialog.createDialog('sync_error', {
+              titleL10nId: errorIds[0],
+              messageL10nId: errorIds[1]
+            }).then(
+              () => {
                 this.showScreen(DISABLED);
-              });
-            });
+              }
+            );
```
Comment (weilonge) [inline tv_apps/browser/js/sync/settings.js]: nit. this arrangement is preferred.
```javascript
}).then(() => {
  this.showScreen(DISABLED);
});
```

File: tv_apps/browser/test/unit/sync/settings_test.js [hunk 4]
```diff
@@ -25,15 +27,21 @@ require('/shared/test/unit/mocks/mock_l20n.js');
...
 
 var mocksForSettings = new MocksHelper([
   'LazyLoader',
   'NavigatorSettings',
   'SettingsListener',
-  'SyncManagerBridge'
+  'SyncManagerBridge',
+  'Browser',
+  'FxosTvModalDialog',
+  'FxosTvInputDialog'
```
Comment (weilonge) [inline tv_apps/browser/test/unit/sync/settings_test.js]: nit. The sorted list would be preferred.

File: tv_apps/browser/js/sync/settings.js (hunk 3 above)
Comment (danhuang1202) [inline tv_apps/browser/js/sync/settings.js]: updated !

File: tv_apps/browser/test/unit/sync/settings_test.js (hunk 4 above)
Comment (danhuang1202) [inline tv_apps/browser/test/unit/sync/settings_test.js]: updated!

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6f7a3327da0e185ea79982d3ac27ba1b87c59d45

**Pull request merged**